File: src/network/packet/impl/ModuleSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\cache\impl\InGameModuleCache;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ModuleSyncPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(private readonly array $data = []) {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->data);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getData(): array {
        return $this->data;
    }

    public static function create(array $data): self {
        return new self($data);
    }

    public static function fromModuleCache(): self {
        return new self(InGameModuleCache::getModuleStates());
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/ModuleSyncPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.cache.ActiveInGameModuleCache;
import de.pocketcloud.cloud.cache.LocalCache;
import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import java.util.*;

@NoArgsConstructor
@Getter
public final class ModuleSyncPacket extends CloudPacket implements ClientboundPacket {

    private List<String> enabledModules;

    public ModuleSyncPacket(List<String> enabledModules) {
        this.enabledModules = enabledModules != null ? enabledModules : List.of();
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(enabledModules);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static ModuleSyncPacket create(List<String> enabledModules) {
        return new ModuleSyncPacket(enabledModules);
    }

    public static ModuleSyncPacket fromModuleCache() {
        return new ModuleSyncPacket(LocalCache.get(ActiveInGameModuleCache.class).getAll().stream().toList());
    }
}